### ISSUE TITLE
sudo nixos-rebuild switch --upgrade: trollius-2.2.post1 not supported for interpreter python3.8

### ISSUE THREAD

**Issue by rldp** [NixOS/nixpkgs#116950]
**Describe the bug**
the command `sudo nixos-rebuild switch --upgrade` fails with the error message `trollius-2.2.post1 not supported for interpreter python3.8`

**To Reproduce**
sudo nixos-rebuild switch --upgrade

**Expected behavior**
no error

**Screenshots**
(none, but see below)

**Additional context**
I ran the command again with the suggested additional option `--show-trace`. Here is the output:

```
unpacking channels...
building Nix...
building the system configuration...
error: while evaluating the attribute 'activationScript' of the derivation 'nixos-system-nixos-21.05pre276515.1f77a4c8c74' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/activation/top-level.nix:95:5:
while evaluating the attribute 'system.activationScripts.script' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/activation/activation-script.nix:80:9:
while evaluating 'textClosureMap' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/strings-with-deps.nix:75:35, called from /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/activation/activation-script.nix:101:18:
while evaluating 'id' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/trivial.nix:14:5, called from undefined position:
while evaluating the attribute 'text' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/activation/activation-script.nix:9:5:
while evaluating the attribute 'text' at undefined position:
while evaluating 'g' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:294:19, called from undefined position:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:139:72, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:297:20:
while evaluating the attribute 'value' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:512:9:
while evaluating the option `system.activationScripts.etc.text':
while evaluating the attribute 'mergedValue' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:544:5:
while evaluating the attribute 'values' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:538:9:
while evaluating the attribute 'values' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:637:7:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:524:28, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:524:17:
while evaluating definitions from `/nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/etc/etc.nix':
while evaluating 'dischargeProperties' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:596:25, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:525:137:
while evaluating the attribute 'value' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:379:44:
while evaluating the attribute 'sources' of the derivation 'etc' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/etc/etc.nix:12:5:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/etc/etc.nix:20:20, called from undefined position:
while evaluating the attribute 'source' at undefined position:
while evaluating 'g' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:294:19, called from undefined position:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:139:72, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:297:20:
while evaluating the attribute 'value' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:512:9:
while evaluating the option `environment.etc.dbus-1.source':
while evaluating the attribute 'mergedValue' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:544:5:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:546:17, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:546:12:
while evaluating 'check' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/types.nix:345:15, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:546:22:
while evaluating the attribute 'serviceDirectories' of the derivation 'dbus-1' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/types.nix:354:14, called from undefined position:
while evaluating the attribute 'value' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:557:27:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:546:17, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:546:12:
while evaluating 'check' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/types.nix:345:15, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:546:22:
while evaluating the attribute 'passAsFile' of the derivation 'system-path' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'out.outPath' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/lib/customisation.nix:156:13:
while evaluating the attribute 'propagatedBuildInputs' of the derivation 'vorta-0.7.5' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/pkgs/development/interpreters/python/mk-python-derivation.nix:111:5:
while evaluating the attribute 'out.outPath' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/lib/customisation.nix:156:13:
while evaluating the attribute 'nativeBuildInputs' of the derivation 'python3.8-APScheduler-3.7.0' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/pkgs/development/interpreters/python/mk-python-derivation.nix:111:5:
while evaluating 'getOutput' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/lib/attrsets.nix:482:23, called from undefined position:
while evaluating 'callPackageWith' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/lib/customisation.nix:117:35, called from /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/pkgs/top-level/python-packages.nix:8326:14:
while evaluating 'makeOverridable' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/lib/customisation.nix:67:24, called from /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/lib/customisation.nix:121:8:
while evaluating anonymous function at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/pkgs/development/python-modules/trollius/default.nix:1:1, called from /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/lib/customisation.nix:69:16:
while evaluating 'makeOverridablePythonPackage' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/pkgs/top-level/python-packages.nix:29:37, called from /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/pkgs/development/python-modules/trollius/default.nix:3:1:
while evaluating 'makeOverridable' at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/lib/customisation.nix:67:24, called from /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/pkgs/top-level/python-packages.nix:31:12:
while evaluating anonymous function at /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/pkgs/development/interpreters/python/mk-python-derivation.nix:30:1, called from /nix/store/i2089cgr8xh7h84vlc7n7k00d4sgm0nq-nixos-21.05pre276515.1f77a4c8c74/nixos/lib/customisation.nix:69:16:
trollius-2.2.post1 not supported for interpreter python3.8
```

**Notify maintainers**

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

> $ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.100, NixOS, 21.05pre275232.29b0d4d0b60 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-21.05pre276515.1f77a4c8c74, nixos-unstable-small-21.03pre262243.9ff1aa8d2ce"`
 - channels(oo): `"nixpkgs-21.03pre269485.a58a0b5098f"`
 - nixpkgs: `/home/oo/.nix-defexpr/channels/nixpkgs`

**Comment by wamserma** [comment]
`trollius` is orphaned by upstream and unmaintained. (It was a Python 2.7-only package.) Unfortunately APScheduler seems to pull this in although it is not required on Python 3.8. (There is a check with with isPy3k, though.)
See `pkgs/development/python-modules/APScheduler/default.nix`

**Comment by veprbl** [comment]
Not broken here:
```
# nix-build channel:nixpkgs-unstable -A python3Packages.APScheduler
this path will be fetched (0.07 MiB download, 0.36 MiB unpacked):
  /nix/store/z8kma27nak1wg19574fsjs76jl9qlkqb-python3.8-APScheduler-3.7.0
copying path '/nix/store/z8kma27nak1wg19574fsjs76jl9qlkqb-python3.8-APScheduler-3.7.0' from 'https://cache.nixos.org'...
/nix/store/z8kma27nak1wg19574fsjs76jl9qlkqb-python3.8-APScheduler-3.7.0
```

**Comment by wamserma** [comment]
Yes, but but there is no Python Version that is supported by trollius and current APScheduler, so @rldp might have some strange customization that leads to this issue.

Maybe sth. like
```
nix-shell -p graphviz imagemagick
export NIXPKGS_ALLOW_BROKEN=1
nix-store -q --graph $(nix-instantiate -I nixos-config=./configuration.nix -E '(with import <nixpkgs/nixos> { }; system)') | dijkstra -da $(nix-instantiate -I nixos-config=./configuration.nix -E '(with import <nixpkgs/nixos> { }; pkgs.python38Packages.trollius)') | gvpr -c 'N[dist>1000.0]{delete(NULL, $)}' | dot -Tsvg | display
```
might provide some insight.

**Comment by veprbl** [comment]
The nixpkgs-unstable in the issue is using 29b0d4d0b60, the fix has landed after that in c75c92f536ed32a83081a4725fd77b3162a07c9f. The 20.09 is not affected because there APScheduler has `doCheck = false`.
